src/oss/langgraph/test.mdx

```diff
@@ -1,3 +1,320 @@
 ---
-title: (Coming Soon) Test
+title: Test
 ---
+
+import AlphaCallout from '/snippets/alpha-lc-callout.mdx';
+
+<AlphaCallout />
+
+After you've prototyped your LangGraph agent, a natural next step is to add tests. This guide covers some useful patterns you can use when writing unit tests.
+
+Note that this guide is LangGraph-specific and covers scenarios around graphs with custom structures - if you are just getting started, check out [this section that uses LangChain prebuilt agents instead](/langchain/test/).
+
+## Prerequisites
+
+:::python
+First, make sure you have [`pytest`](https://docs.pytest.org/) installed:
+
+```bash
+$ pip install -U pytest
+```
+:::
+
+:::js
+First, make sure you have [`vitest`](https://vitest.dev/) installed:
+
+```bash
+$ npm install -D vitest
+```
+:::
+
+## Getting started
+
+Because many LangGraph agents depend on state, a useful pattern is to create your graph before each test where you use it, then compile it within tests with a new checkpointer instance.
+
+The below example shows how this works with a simple, linear graph that progress from `node1` to `node4`. Each node updates a single state key:
+
+:::python
+```python
+import pytest
+
+from typing_extensions import TypedDict
+from langgraph.graph import StateGraph, START, END
+from langgraph.checkpoint.memory import MemorySaver
+
+@pytest.fixture
```

Comment:
I would suggest not using a fixture in this case.

* A fixture is a more advanced topic that some users will not know.
* In reality, users do not need to use a fixture. They will likely have some code in their application that creates a graph or maybe even a global instance, and they'll just import it in the test code.

So you could do this:

```python
def create_graph() -> StateGraph:
  ....
  return graph
```